pkg/entrypoint/entrypointer.go

```diff
@@ -121,6 +123,25 @@ func (e Entrypointer) Go() error {
 	return err
 }
 
+// fmtErrOutPut will format error to PipelineResourceResult
+func (e Entrypointer) fmtErrOutPut(err error) []v1alpha1.PipelineResourceResult {
+	if err == nil {
+		return []v1alpha1.PipelineResourceResult{}
+	}
+	// TODO: will add logic to format reason according error type sooner, example: DeadlineExceeded
+
+	return []v1alpha1.PipelineResourceResult{
+		{
+			Key:   "Reason",
+			Value: err.Error(),
+		},
+		{
```

Comment:
Why do we need both?

Reply:
`errors` will be reflected on `Status` of `TaskRun`.
So, about the scene of `timedout` https://github.com/tektoncd/pipeline/issues/1690,
 we should expect the terminated reason is 
 `Timeout` and  message is `xxx step is timeout more than %s`.

`Tye` is a convenient way to do some programming judgement.
`Message` is friendly information to help people to do some troubleshootings.

Sometimes the type is clearly like `Timeout` or `Canceled`  but not all cases. The error type is not known for us, eg.  just binary executed error. In this scenario，I assign `err.Error()` to `Reason`, May be `NonZeroExit` is another choice ？

Reply:
Reason should be a short machine-consumable `CamelCaseString`, while Message should be a long human-readable sentence.

Reply:
I still see `err.Error()` as both Reason and Message, can we have a short `CamelCaseString` for Reason?

Reply:
Sorry , I did not realize that I should change the Reason in last comment.
I have changed the Reason to `Unknown` as default .

Reply:
> I still see `err.Error()` as both Reason and Message, can we have a short `CamelCaseString` for Reason?

Sorry , I did not realize that I should change the Reason in last comment.
I have changed the Reason to `Unknown` as default.